### PR TITLE
Monk conversions instead of killings

### LINKED ISSUE #375: Monk kills enemy unit instead of converting to own team
and he doesn't keep distance to enemy unit.

### PR DESCRIPTION
**Pull request by LordAro**
Supercedes #428. No functional changes (I haven't even tested it beyond compilation & tests!), and @bombadie retains primary commit authorship

I got very bored.

Fixes #375

### CODE REVIEW

**Comment by MaanooAk** [comment]
There are more things that need to be done.
It you change only the owner upgrades will keep applying and the population tracker and the player base triggers will not work.
I believe the right method would be to:
- remove the unit
- create a copy of the unit type (find a way to not create a lot of copies)
- add a new unit (through the new owener player)

**Comment by LordAro** [comment]
Yeah, figured it wouldn't be so easy :)

I'll take a look and see what I can do (probably not very soon though)

File: libopenage/unit/unit.h [hunk 1]
```diff
@@ -175,6 +175,12 @@ class Unit : public log::LogSource {
 	void add_attribute(std::shared_ptr<AttributeContainer> attr);
 
 	/**
+	 * delete a given attribute this this unit
+	 * this is used to delete/change things like color, hitpoints and speed
```
Comment (castilma) [inline libopenage/unit/unit.h]:
```
Delete a given attribute from this unit.
^                        ^~~~
```
```
This is used to delete/change things like color, hitpoints and speed.
^
```
Thank you.

Comment (LordAro) [inline libopenage/unit/unit.h]: I noticed this at the time, but (most of) the rest of the doccomments in this file have a similar lack of punctuation. something to fix elsewhere perhaps?

Comment (TheJJ) [inline libopenage/unit/unit.h]: If you like, everywhere :P

Comment (LordAro) [inline libopenage/unit/unit.h]: I meant in a different prq ;)

Comment (TheJJ) [inline libopenage/unit/unit.h]: me too :D

Comment (LordAro) [inline libopenage/unit/unit.h]: deferred to @castilma in #832 :p

Comment (castilma) [inline libopenage/unit/unit.h]: hey!

**Comment by simonsan** [comment]
Will most likely be superseded by new gamestate implementation. Will mark it as `zombie` for later usages though. Thank you.

**Pull request closed without merging**